### PR TITLE
NIAD-1154: Deploy GPCC adapter service 

### PR DESCRIPTION
**Pull request by RIashfaqahmed**
2 new components are adding

* gpc-consumer
* gpc-consumer-wriemock

### CODE REVIEW

File: terraform/aws/components/gpc-consumer/locals.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  resource_prefix = "${var.project}-${var.environment}-${var.component}"
+
+  availability_zones = ["${var.region}a", "${var.region}b", "${var.region}c"]
+
+  
+  # Use below when the ECR repo is created by terraform in account component.
+  #image_name = "${data.terraform_remote_state.account.outputs.nhais_ecr_repo_url}:${var.nhais_build_id}"
+  image_name = "${var.account_id}.dkr.ecr.${var.region}.amazonaws.com/gpc-consumer:${var.gpc-consumer_build_id}"
+  wiremock_image_name = "${var.account_id}.dkr.ecr.${var.region}.amazonaws.com/gpc-consumer-wiremock:${var.gpc-consumer_build_id}"
```
Comment (LukaszKwasniewski) [inline terraform/aws/components/gpc-consumer/locals.tf:16]: is wiremock needed for gpc?

File: terraform/aws/components/gpc-consumer/locals_env_variables.tf [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    {
+      name  = "GPC_CONSUMER_SERVER_PORT"
+      value = var.gpc-consumer_service_container_port
+    },
+    {
+      name  = "GPC_CONSUMER_ROOT_LOGGING_LEVEL"
+      value = var.gpc-consumer_root_log_level
+    },
+    {
+      name  = "GPC_CONSUMER_ROOT_LOGGING_LEVEL"
```
Comment (LukaszKwasniewski) [inline terraform/aws/components/gpc-consumer/locals_env_variables.tf]: name of the env variable is the same as one above it, but uses different value. Which one is the correct one?

File: terraform/aws/components/gpc-consumer/locals.tf (hunk 1 above)
Comment (RIashfaqahmed) [reply terraform/aws/components/gpc-consumer/locals.tf:16]: Matt mentioned it in https://gpitbjss.atlassian.net/browse/NIAD-1154

File: terraform/aws/components/gpc-consumer/locals_env_variables.tf [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      value = var.gpc-consumer_log_level
+    },
+    {
+      name  = "GPC_CONSUMER_LOGGING_FORMAT"
+      value = var.gpc-consumer_logging_format
+    },
+    {
+      name  = "GPC_CONSUMER_URL"
+      value = "http://${module.gpc-consumer_ecs_service[0].loadbalancer_dns_name}:${var.gpc-consumer_service_container_port}"
```
Comment (LukaszKwasniewski) [inline terraform/aws/components/gpc-consumer/locals_env_variables.tf]: as the gpc-consumer module does not have `count` on it the [0] index is not needed here.

File: terraform/aws/components/gpc-consumer/locals_env_variables.tf (hunk 2 above)
Comment (RIashfaqahmed) [reply terraform/aws/components/gpc-consumer/locals_env_variables.tf]: in https://github.com/nhsconnect/integration-adaptor-gpc-consumer/blob/main/README.md there are two logging env variables. It was a typo and now fixed.

GPC_CONSUMER_ROOT_LOGGING_LEVEL | WARN | The logging level applied to the entire application (including third-party dependencies).

GPC_CONSUMER_LOGGING_LEVEL | INFO | The logging level applied to GPC Consumer Adaptor components.

File: terraform/aws/components/gpc-consumer/locals_env_variables.tf (hunk 3 above)
Comment (RIashfaqahmed) [reply terraform/aws/components/gpc-consumer/locals_env_variables.tf]: removed [0] index 

    {
      name  = "GPC_CONSUMER_URL"
      value = "http://${module.gpc-consumer_ecs_service.loadbalancer_dns_name}:${var.gpc-consumer_service_container_port}"
    },

**Comment by RIashfaqahmed** [comment]
Devs decided to deploy gpc-consumer adapter through GP2GP repo. So closing this PR.

**Pull request closed without merging**